app/vmock.go

```diff
@@ -296,3 +314,36 @@ func handleVMockDuty(ctx context.Context, duty core.Duty, eth2Cl eth2wrap.Client
 
 	return nil
 }
+
+// newRegistrations returns a list of validator registrations for the given cluster.
+func newRegistrations(cState state.Cluster) ([]*eth2api.VersionedValidatorRegistration, error) {
+	timestamp, err := eth2util.ForkVersionToGenesisTime(cState.ForkVersion)
+	if err != nil {
+		return nil, err
+	}
+	timestamp = timestamp.Add(time.Minute) // Use slot>0 to override pre-generated registrations.
```

Comment:
this will be incompatible with teku VC, if we use vmock in a cluster with teku VCs. I'd suggest to use teku proposer endpoint in vmock to be compatible with teku registrations.

Reply:
cool, switched to fetching registrations from vapi